### ISSUE TITLE
"Mail not sent; to enable sending, set the MAIL_URL environment variable." despite having set MAIL_URL

### ISSUE THREAD

**Issue by khba** [wekan/wekan#1072]
Hello,
After modifying our wekan sender mail from wekan2@ourdomain.com to wekan@oudomain.com, mails are no longer sent : 

```
[root@dockerhost wekan-mongodb]# docker-compose logs -f wekan
Attaching to wekan-app
wekan-app  | { Error: Cannot find module '../build/Release/bson'
wekan-app  |     at Function.Module._resolveFilename (module.js:469:15)
wekan-app  |     at Function.Module._load (module.js:417:25)
wekan-app  |     at Module.require (module.js:497:17)
wekan-app  |     at require (internal/module.js:20:19)
wekan-app  |     at Object.<anonymous> (/build/programs/server/npm/node_modules/meteor/cfs_gridfs/node_modules/mongodb/node_modules/bson/ext/index.js:15:10)
wekan-app  |     at Module._compile (module.js:570:32)
wekan-app  |     at Object.Module._extensions..js (module.js:579:10)
wekan-app  |     at Module.load (module.js:487:32)
wekan-app  |     at tryModuleLoad (module.js:446:12)
wekan-app  |     at Function.Module._load (module.js:438:3) code: 'MODULE_NOT_FOUND' }
wekan-app  | js-bson: Failed to load c++ bson extension, using pure JS version
wekan-app  | Note: you are using a pure-JavaScript implementation of bcrypt.
wekan-app  | While this implementation will work correctly, it is known to be
wekan-app  | approximately three times slower than the native implementation.
wekan-app  | In order to use the native implementation instead, run
wekan-app  | 
wekan-app  |   meteor npm install --save bcrypt
wekan-app  | 
wekan-app  | in the root directory of your application.
wekan-app  | ====== BEGIN MAIL #0 ======
wekan-app  | (Mail not sent; to enable sending, set the MAIL_URL environment variable.)
wekan-app  | Content-Type: text/plain
wekan-app  | From: wekan@ourdomain.com
wekan-app  | To: mymail@ourdomain.com
wekan-app  | Subject: [Wekan] Activity Notification
wekan-app  | Message-ID: <7f27e313-7733-c14e-8150-d8328fc6f281@sfr.fr>
wekan-app  | Content-Transfer-Encoding: 7bit
wekan-app  | Date: Thu, 15 Jun 2017 11:41:15 +0000
wekan-app  | MIME-Version: 1.0
wekan-app  | 
wekan-app  | John Doe added "new2" to "WEKAN"
wekan-app  | https://wekanhost4/b/DS7YJ7kPcBBknSzD7/wekan
			====== END MAIL #0 ======
wekan-app  | { Error: Cannot find module '../build/Release/bson'
wekan-app  |     at Function.Module._resolveFilename (module.js:469:15)
wekan-app  |     at Function.Module._load (module.js:417:25)
wekan-app  |     at Module.require (module.js:497:17)
wekan-app  |     at require (internal/module.js:20:19)
wekan-app  |     at Object.<anonymous> (/build/programs/server/npm/node_modules/meteor/cfs_gridfs/node_modules/mongodb/node_modules/bson/ext/index.js:15:10)
wekan-app  |     at Module._compile (module.js:570:32)
wekan-app  |     at Object.Module._extensions..js (module.js:579:10)
wekan-app  |     at Module.load (module.js:487:32)
wekan-app  |     at tryModuleLoad (module.js:446:12)
wekan-app  |     at Function.Module._load (module.js:438:3) code: 'MODULE_NOT_FOUND' }
wekan-app  | js-bson: Failed to load c++ bson extension, using pure JS version
wekan-app  | Note: you are using a pure-JavaScript implementation of bcrypt.
wekan-app  | While this implementation will work correctly, it is known to be
wekan-app  | approximately three times slower than the native implementation.
wekan-app  | In order to use the native implementation instead, run
wekan-app  | 
wekan-app  |   meteor npm install --save bcrypt
wekan-app  | 
wekan-app  | in the root directory of your application.

```
The "MAIL_URL environment variable" has been set both on docker-compose.yml file" and also on the shell. So why dos it keep saying: "(Mail not sent; to enable sending, set the MAIL_URL environment variable.)" ???

And also things used to work before changing the sender address. So I changed back the sender address to what it was but still, mails aren't going.  

Why arent't the mails going? 

The server's response is: "550 5.7.1 Message content rejected (in reply to end of DATA command))"  to mails fro wekan, whereas when I send the same mails manually with telnet/netcat it works. 

About the line "meteor npm install --save bcrypt", not sure if it's related but I installed npm  and did:
    npm install --save bcrypt
on several directories but keep having the errors. How and where do I install this bcrypt library knowing that I'm using the wekan docker image and the "host" server doesn't know the command "meteor" nor the package "meteor" when I do  "yum install meteor".?

Thanks,

**Comment by xet7** [comment]
@khba 

Do you use latest Wekan docker-mongodb ? Have you checked Troubleshooting Mail page?
https://github.com/wekan/wekan/wiki/Platforms

**Comment by khba** [comment]
I moved to the last docker image, now the wekan app logs nothing when I send a mail. 
The mails are delivered sometimes and rejected some other times with the postfix logs "Mail content rejected in reply to End Of Data command".

**Comment by xet7** [comment]
@khba 

Does postfix reject mail, or some outside server where you are sending mail ?

**Comment by xet7** [comment]
@khba 

Usually in email settings, if Wekan logs show nothing, the error is not in Wekan, but in some other system.

**Comment by khba** [comment]
No the SMTP relay server is the one rejecting the mails. The weird part is that before changing the sender adress from wekan2@ourdomain.com to wekan@ourdomain.com the mails were being correctly delivered.

**Comment by xet7** [comment]
@khba 

You could check are email addresses same both at environment variables and Admin Panel.

And if you change email address back to wekan2@yourdomain.com , does this fix those problems?

**Comment by khba** [comment]
Yes they're the same and no setting back the first email used didn't fix the problem. 

And to have the mail changing take effect, a restart isn't enough, I have to remove and recreate the containers with docker-compose down and up again, but that's just information.

**Comment by xet7** [comment]
@khba 

Can you update to today's latest version of Docker container and check are there still problems?

**Comment by khba** [comment]
I did, still the same:

Postfix logs: 
Jun 20 11:48:37 dockerhost postfix/qmgr[1636]: 2977311F77A4: from=<wekan2@ourdomain.com>, size=564, nrcpt=1 (queue active)
Jun 20 11:48:37 dockerhost postfix/smtp[11721]: 2977311F77A4: to=<mymail@ourdomain.com>, relay=smtp.ourdomain.com[X.X.X.X]:25, delay=0.16, delays=0.06/0.01/0.03/0.06, dsn=5.7.1, status=bounced (host smtp.ourdomain.com[X.X.X.X] said: 550 5.7.1 Message content rejected (in reply to end of DATA command))

Wekan app logs nothing.

**Comment by xet7** [comment]
@khba 

Do you have spam filtering there, that could reject message?

Or is it possible to get the actual message content captured somehow to see why it was rejected?

What language you you use in Wekan, could it complain about some unicode characters?

**Comment by xet7** [comment]
@khba 

From those logs it looks like that Wekan sent email successfully, but postfix bounced message because of some rule.

**Comment by xet7** [comment]
@khba 

I don't know could there be some blacklist or some other email server related issue.

**Comment by xet7** [comment]
@khba 

Some email servers require SPF records to ensure deliverability.

**Comment by khba** [comment]
About the spam filtering it's possible, I contacted the postmaster of our smtp.ourdomain.com without getting answer. But the mails used to be delivered, unless they changed their policy in the meantime. 

We use french, yes I did a wireshark capture of the mail and compared it to when I send the mail manually with netcat (in that case the mail is delivered, with the same mail content, sender and receiver addresses), the two captures look alike except that wekan uses ESMTP protocol to send the mail while with netcat it's SMTP. But I don't think it's a pb since mails used to go anyway. 

Okay looking into that SPF thing.
